publish.gradle

```diff
@@ -49,6 +50,18 @@ if (!hasProperty('releaseType')) {
     }
 }
 
+if (project.hasProperty("ntPublishVersion")) {
```

Comment:
Could you move this to just under where you define ntVersion? Otherwise, on first glance it looks like a use-before-assignment bug. It's not, but it hurts anyway.